File: data/gpt-4o/readable_format_HPEC/cpp/file085.cpp
#include <stdio.h>
#include <stdlib.h>
#include <omp.h>
#include <math.h>

double a[100];
int main()
{
 int i, len = 100, rem, j;
 double u;

 for (i = 0; i < len; i++)
 {
  a[i] = i+1;
 }

 u = (double)rand() / RAND_MAX;
 j = floor(100*u);

#pragma omp parallel for if (j%2==0)
 for (i = 0; i < len-1; i++)
 {
  a[i+1] = a[i]+1;
 }

 printf("a[50] = %.f
",a[50]);

 return 0;
}

File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file085.f90
program DRB114_if_orig_yes
    use omp_lib
    implicit none

    integer i, len, rem, j
    real :: u
    integer, parameter :: dp = kind(1.0d0)
    real(dp), dimension(:), allocatable :: a

    len = 100
    allocate (a(len))

    do i = 1, len
        a(i) = i
    end do

    call random_number(u)
    j = FLOOR(100*u)

    !$omp parallel do if (MOD(j,2)==0)
    do i = 1, len-1
        a(i+1) = a(i)+1
    end do
    !$omp end parallel do

    print*,'a(50) =',a(50)

    deallocate(a)
end program
